api/src/main/java/org/openmrs/FormField.java

```diff
@@ -23,33 +35,55 @@
  * @see org.openmrs.Form
  * @see org.openmrs.Field
  */
+@Entity
+@Table(name = "form_field")
 @Audited
 public class FormField extends BaseChangeableOpenmrsMetadata implements java.io.Serializable, Comparable<FormField> {
 	
 	public static final long serialVersionUID = 3456L;
 	
 	// Fields
-	
+	@Id
+	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "form_field_id_seq")
+	@GenericGenerator(
+			name = "form_field_id_seq",
+			strategy = "native",
+			parameters = @Parameter(name = "sequence", value = "form_field_form_field_id_seq")
+		)
+	@Column(name = "form_field_id")
 	protected Integer formFieldId;
 	
+	@ManyToOne
+	@JoinColumn(name = "parent_form_field")
 	protected FormField parent;
 	
+	@ManyToOne
+	@JoinColumn(name = "form_id", nullable = false)
 	protected Form form;
 	
+	@ManyToOne(cascade = CascadeType.ALL)
```

Comment:
Is this `CascadeType` correct?

Reply:
It has cascade="save-update,merge,evict" in hibernate.xml file

Reply:
`CascadeType.ALL` includes all cascade operations `(PERSIST, REMOVE, REFRESH, MERGE, DETACH)`. Consider specifying only the required cascade types